recipes/arrow/all/conanfile.py

```diff
@@ -392,6 +392,7 @@ def generate(self):
         if is_msvc(self):
             tc.variables["ARROW_USE_STATIC_CRT"] = is_msvc_static_runtime(self)
         tc.variables["ARROW_DEPENDENCY_SOURCE"] = "SYSTEM"
+        tc.variables["ARROW_PACKAGE_KIND"] = "conan"
```

Comment:
What does this do? Oooo

Reply:
`ARROW_PACKAGE_KIND` is a general Apache Arrow C++ build option to add build metadata.

This is used for handling Conan packages specially in this case:
* https://github.com/apache/arrow/blob/master/cpp/cmake_modules/FindjemallocAlt.cmake#L26-L42
* https://github.com/apache/arrow/blob/master/cpp/cmake_modules/FindBrotliAlt.cmake#L24-L53

They are done by patching Apache Arrow C++ (see the deleted patch: https://github.com/conan-io/conan-center-index/pull/14903/files#diff-021dc424ccea95d24c75533b53af25e42bb6e51b1ef86d95b1b2c51d3d017ed3 ) before this change. But it increasing maintenance cost of this Conan package. We don't need to patch this Conan package with this approach.

Reply:
Ooooo thank you for the amazing answer ❤️ 

Now can we please stick it in as a comment for future contributions/reviwers? 🙏 

```suggestion
        tc.variables["ARROW_PACKAGE_KIND"] = "conan" # See https://github.com/conan-io/conan-center-index/pull/14903/files#r1057938314 for details
```

Reply:
Sure.
I've applied your suggestion.